src/environment.cpp

```diff
@@ -1064,7 +1064,8 @@ void ServerEnvironment::step(float dtime)
 	/*
 		Manage active block list
 	*/
-	if(m_active_blocks_management_interval.step(dtime, 2.0))
+	float abm_interval = g_settings->getFloat("abm_interval");
```

Comment:
setting getter has a bad performance. Caching is required, especially on ServerEnv::step.
Also modify the abm_interval at runtime can be dangerous